src/ILICheck.Web/Controllers/UploadController.cs

```diff
@@ -218,73 +235,108 @@ private async Task<IActionResult> UploadToDirectoryAsync(HttpRequest request)
+                                        await hubContext.Clients.Client(connectionId).SendAsync("validationAborted", "Mehrere Transferdateien gefunden!" + uploadInstructionMessage);
+                                        LogInfo("Validation aborted, .zip-archive contains several transfer files.");
+                                        mainCts.Cancel();
+                                        return;
+                                    }
+                                }
+
+                                isInterlis2 = transferfileExtension == ".xtf";
+
+                                if (string.IsNullOrEmpty(transferfileExtension))
+                                {
+                                    await hubContext.Clients.Client(connectionId).SendAsync("validationAborted", "Fehlende Datei(en)!" + uploadInstructionMessage);
+                                    LogInfo(".zip-archive does not contain required files.");
+                                    mainCts.Cancel();
+                                    return;
+                                }
+
+                                if (Path.GetFullPath(zipFilePath).StartsWith(extractPath, StringComparison.Ordinal))
+                                {
+                                    var parentDirectory = Directory.GetParent(zipFilePath).FullName;
+                                    archive.Entries.ToList().ForEach(entry => entry.ExtractToFile(Path.Combine(parentDirectory, entry.Name)));
+
+                                    // check if multiple transferfiles exsist in directory
+                                    var transferfiles = Directory.GetFiles(parentDirectory).Where(file => Path.GetExtension(file).ToLower() == transferfileExtension);
+                                    unzippedTransferfilePath = transferfiles.Single(file => Path.GetExtension(file).ToLower() == transferfileExtension);
+                                }
+                                else
+                                {
+                                    await hubContext.Clients.Client(connectionId).SendAsync("validationAborted", "Dateipfad konnte nicht aufgelöst werden!" + uploadInstructionMessage);
+                                    LogInfo("Upload aborted, cannot get extraction path.");
+                                    mainCts.Cancel();
+                                    return;
+                                }
                             }
                             else
                             {
-                                await hubContext.Clients.Client(connectionId).SendAsync("validationAborted", "Dateipfad konnte nicht aufgelöst werden.");
-                                LogInfo("Upload aborted, cannot get extraction path.");
+                                await hubContext.Clients.Client(connectionId).SendAsync("validationAborted", "Nicht unterstützte Dateien, bitte laden Sie ausschliesslich .xtf, .ift und .xml hoch! " + uploadInstructionMessage);
```

Comment:
Q: .ili noch ergänzen?
```suggestion
                                await hubContext.Clients.Client(connectionId).SendAsync("validationAborted", "Nicht unterstützte Dateien, bitte laden Sie ausschliesslich .xtf, .ift, .ili und .xml hoch! " + uploadInstructionMessage);
```